packages/netlify-cms-core/src/actions/entries.js

```diff
@@ -451,6 +452,14 @@ export function createEmptyDraftData(fields, withNameKey = true) {
       return acc;
     }
 
+    // A boolean field without a default value defined will initially render
+    // render as false, but the field will fail the required check. So we are
+    // assinging false.
+    if (boolean && defaultValue === null) {
+      acc[name] = false;
+      return acc;
```

Comment:
@chrfritsch  It's best not to return or assign to `acc` here because the `withNameKey` check will not be honored. I suggest you simply change the default value to `false` inside the `if` check or use this onliner `boolean && defaultValue === null && (defaultValue = false);` if it's readable.

Reply:
Changing the defaultValue would introduce more changes because it's declared as const . So I think I came around with an even better solution.